### PR TITLE
Added fittemplate to ft_prepare_mesh

### PR DESCRIPTION
**Pull request by simonhomoelle**
Fittemplate allows ft_prepare_mesh to transform a mesh on the basis of a surface scan and a template head  model

### CODE REVIEW

File: test/test_bug3473.m [hunk 1]
```diff
@@ -0,0 +1,95 @@
+function test_bug3453
```
Comment (robertoostenveld) [inline test/test_bug3473.m]: test scripts should have WALLTIME and MEM specifications at the top.

File: ft_prepare_mesh.m [hunk 2]
```diff
@@ -16,7 +16,8 @@
 %
 % Configuration options:
 %   cfg.method      = string, can be 'interactive', 'projectmesh', 'iso2mesh', 'isosurface',
-%                     'headshape', 'hexahedral', 'tetrahedral', 'cortexhull'
+%                     'headshape', 'hexahedral', 'tetrahedral',
+%                     'cortexhull', 'fittemplate'
```
Comment (robertoostenveld) [inline ft_prepare_mesh.m]: please keep them on a single line (i.e. 2 in total, not 3)

File: private/prepare_mesh_fittemplate.m [hunk 3]
```diff
@@ -0,0 +1,48 @@
+function M = prepare_mesh_fittemplate(headshape,template)
+
+% PREPARE_MESH_FITTEMPLATE computes an affine transformation matrix between 2 point clouds 
+%
+% This function relies on cpd toolbox found in the external/cpd folder
```
Comment (robertoostenveld) [inline private/prepare_mesh_fittemplate.m]: Please provide a link (URL) to the toolbox.

File: private/prepare_mesh_fittemplate.m [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+%
+%    You should have received a copy of the GNU General Public License
+%    along with FieldTrip. If not, see <http://www.gnu.org/licenses/>.
+%
+% $Id$
+
+% ensure that the input is consistent with what this function expects
+
+% add toolbox cpd
+ft_hastoolbox('cpd',2);
```
Comment (robertoostenveld) [inline private/prepare_mesh_fittemplate.m]: add a space behind the comma

File: private/prepare_mesh_fittemplate.m [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+opt.corresp = 0;
+opt.method  = 'affine';
+opt.max_it = 100;
+opt.fgt=0;
```
Comment (robertoostenveld) [inline private/prepare_mesh_fittemplate.m]: add spaces around the "="

File: test/test_bug3473.m [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+%filename = '/home/common/matlab/fieldtrip/data/ftp/tutorial/epilepsy/case1/ctf_data/case1.pos';
+filename = 'D:\Development\Data\case1\ctf_data/case1.pos';
+%% Individualizing a template volume conduction to on the basis of surface information
+%
+%% # Introduction
```
Comment (robertoostenveld) [inline test/test_bug3473.m]: please remove the lengthy comments, unless you think that it is relevant for understanding the script. Better is to point from the script to the corresponding page on the website.

File: utilities/ft_hastoolbox.m [hunk 7]
```diff
@@ -388,6 +389,8 @@
     dependency = {'fig2plotly' 'savejson'};
   case 'JSONIO'
     dependency = {'jsonread', 'jsonwrite', 'jsonread.mexa64'};
+  case 'CPD' 
+    dependency = {'cpd'};
```
Comment (robertoostenveld) [inline utilities/ft_hastoolbox.m]: if possible add some more functions from the toolbox to the list. This one function is not very distinctive.

File: test/test_bug3473.m (hunk 1 above)
Comment (simonhomoelle) [reply test/test_bug3473.m]: Add WALLTIME and MEM to the script

File: ft_prepare_mesh.m (hunk 2 above)
Comment (simonhomoelle) [reply ft_prepare_mesh.m]: Put it back to 2 lines

File: private/prepare_mesh_fittemplate.m (hunk 4 above)
Comment (simonhomoelle) [reply private/prepare_mesh_fittemplate.m]: add space

File: private/prepare_mesh_fittemplate.m (hunk 5 above)
Comment (simonhomoelle) [reply private/prepare_mesh_fittemplate.m]: Spaces added around "="

File: test/test_bug3473.m (hunk 6 above)
Comment (simonhomoelle) [reply test/test_bug3473.m]: The lengthy comments have been removed.

File: utilities/ft_hastoolbox.m (hunk 7 above)
Comment (simonhomoelle) [reply utilities/ft_hastoolbox.m]: I added more functions from the toolbox that are necessary for fittemplate

File: private/prepare_mesh_fittemplate.m (hunk 3 above)
Comment (simonhomoelle) [reply private/prepare_mesh_fittemplate.m]: Added the URL

**Comment by robertoostenveld** [comment]
looks good. let me know when you think it is ready to be merged.

**Comment by simonhomoelle** [comment]
> looks good. let me know when you think it is ready to be merged.

Ready to be merged!

**Comment by robertoostenveld** [comment]
this also solves the bug reported in #975

**Pull request merged**